components/automate-ui/src/app/page-components/insight-attributes-dropdown/insight-attributes-dropdown.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, OnInit, Input, Output, EventEmitter, HostBinding } from '@angular/core';
+import { trigger, transition, style, animate, query, group, animateChild } from '@angular/animations';
+import { FilterOption, FilterableOptions } from './insight-attributes-dropdown.model';
+import { isEqual } from 'lodash/fp';
+
+
+// Animations
+const menuOpenAnim = trigger('menuOpenAnim', [
+  transition(':enter', [
+    style({ opacity: 0, transform: 'translateY(-8px) scale(0)' }),
+    group([
+      query('@childAnim', animateChild()),
+      animate('220ms ease-out', style({ opacity: 1, transform: 'translateY(0px) scale(1)' }))
+    ])
+  ]),
+  transition(':leave', [
+    style({ opacity: 1 }),
+    animate('195ms ease-in', style({ opacity: 0 }))
+  ])
+]);
+
+const childAnim = trigger('childAnim', [
+  transition(':enter', [
+    style({ opacity: 0 }),
+    animate('50ms 170ms', style({ opacity: 1 }))
+  ])
+]);
```

Comment:
Are animations defined/spec'd somewhere or just lookin good (totally valid 😄)? Curious how the various `ms` timing values were chosen.

Not super specific to this PR. Similar to things like fonts and colors, figuring out how we want to ensure animation/transition consistency across components is my main motivation of the question.

Reply:
Hi @scottopherson , I showed a demo of the animation with @SEAjamieD , you can try it yourself here, it works just like the invasion prototype: https://xd.adobe.com/view/1ca55366-2ffc-484a-6bd8-e5834716efe4-6a94/?fullscreen
I haven't circled the design with the entire UX and UI team yet, but just to give you an idea of how it might look like in the future.
The spec I've set up so far is `ease-out` and `ease-in`, `400ms` for each transition.

Reply:
We haven't really had a discussion around specifics for what is getting animated and what is not.  With the desire to animate lots of the desktop coming up, I wanted to learn a little more.  I based the `ms` timing on google's recommendation of `220ms` coming in and `190` going out.  The `50ms 170ms` is for the children, so that they delay for 170 and then fade in quickly in the last 50ms and end at the same time as the parent.

No need to have this go live, can pull it easily or we can add a `[@.disabled]` attribute to disabled the animations for now.

Agreed a larger conversation needs to be had.  One thing that may be nice is we can define the animations in another file and just import and reuse.

Reply:
This article is different than the original one I was looking at, but the concepts are the same.  Out faster than it comes in: https://material.io/design/motion/speed.html#duration

Reply:
thanks for sharing the link @SEAjamieD , it is helpful!
yeah, I agree that we should discuss the specs for animation more in the future. The one I used for the large panels shouldn't apply to every element in the UI (as suggested in the link, too)